### PR TITLE
Consume file icon service

### PR DESCRIPTION
**Pull request by acontreras89**
Consume `atom.file-icons` service to display icons next to file names. This is based on https://github.com/atom/atom/issues/9820 and has been suggested by @lee-dohm. Another example is available in https://github.com/atom/tree-view/pull/490.

Current behavior has been kept.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,12 @@
...
 ![](https://cloud.githubusercontent.com/assets/18362/10862852/c6de2de0-800d-11e5-8158-284f30aaf5d2.png)
+
+## API
+
+Tabs can display icons next to file names. These icons are customizable by installing a package that provides an `atom.file-icons` service.
+
+The `atom.file-icons` service must provide the following methods:
+
+* `iconClassForPath(path)` - Returns a CSS class name to add to the tab view
+* `onWillDeactivate` - An event that lets the tree view return to its default icon strategy
```
Comment (lee-dohm) [inline README.md]: "lets the tabs return to their"

**Comment by acontreras89** [comment]
I added specs and corrected the README. Are me missing anything else?

**Comment by lee-dohm** [comment]
There is a bug in the tree-view version that I was wondering if it might creep in here. The icons wouldn't show up on first load in the tree-view. Can you reproduce that here?

**Comment by acontreras89** [comment]
Neither the `tree-view` implementation nor the `tabs` one were updating the rendered icons when the service is replaced. I'll fix that right away.

**Comment by lee-dohm** [comment]
Thanks for that! :grinning:

**Comment by Alhadis** [comment]
@lee-dohm Any idea when this will make it into a public release? Would love to make use of this in the [File Icons](https://github.com/DanBrooker/file-icons) package. =)

**Comment by lee-dohm** [comment]
Currently it is on the v1.9.x release schedule.

**Comment by Arcanemagus** [comment]
Looks like this made it into Atomv1.7.4 @Alhadis.

**Comment by Alhadis** [comment]
Yup! I noticed. =) One down, [two t](atom/tree-view#825)[o go](https://github.com/atom/tree-view/pull/833).

**Pull request merged**